src/jshint.js

```diff
@@ -1340,8 +1340,10 @@ var JSHINT = (function() {
       while (!obj.identifier && typeof obj.left === "object")
         obj = obj.left;
 
-      if (obj.identifier && natives.indexOf(obj.value) >= 0)
+      if (obj.identifier && natives.indexOf(obj.value) >= 0 &&
+          state.funct["(scope)"].isPredefined(obj.value)) {
```

Comment:
Taking this approach means `findNativePrototype` would no longer be purely functional. `jshint.js` has a lot of implicit dependencies between functions, so I think it's worthwhile to defend the rare cases where function behavior is fully defined by explicit input/output. This is easily resolved, though: just move the check to `checkLeftSideAssign`